### DISCUSSION TITLE
Working in batches?

### DISCUSSION THREAD

**Discussion by robintschoetschel** [asreview/asreview#1460] [Q&A]
Hi there! 

First of all: Great project, I'm very happy to have found it. I've been wondering if there is a way to use ASR in a kind of "batchwise"-mode for "offline" work and/or collaboration. Here's the workflow I imagine.

1. I create a partially labelled dataset, possibly with the help of multiple coders.
2. ASR reads the file and prepares the next batch of N entries to review.
3. I download the file, work on the whole file (again, possibly with multiple coders), and then re-upload the new ratings.
4. ASR reads the new information and prepares a new batch
5. Rinse and repeat until I reach a stopping criterion (e.g. the last batch was all negatives)

Alternatively, I could just prepare a new merged, partially labelled file in a new project and use that to get a new batch for each iteration. 

I'm pretty certain this should be possible somehow. As far as I understand, ASR creates a ranking of the dataset based on existing information — maybe I can use the API to access the ordered dataset?

Thanks for your help!

**Comment by robintschoetschel** [comment]
I just found that the export includes an asreview_ranking column that I can use for that purpose. Except for convenience (I have to recreate projects every time I do a new batch), does anything speak against this procedure? 

Maybe this can also be seen as a feature request :)

**Comment by Rensvandeschoot** [comment]
These are great thoughts, and I appreciate your proactive approach to using ASReview. To delve into your query in more depth, I'll respond substantively in due time. 

In the interim, I highly recommend you take a look at this article: ["Reproducibility and Data storage for Active Learning-Aided Systematic Reviews"](https://psyarxiv.com/g93zf). It provides valuable insight into our data storage practices and may answer some of the questions you have.

Moreover, if you're interested in accessing additional information from your ASReview files, our [documentation](https://asreview.readthedocs.io/en/latest/example_api_asreview_file.html) provides a comprehensive guide to doing so. This should help you gain more control over your data and make the most out of your systematic review process.

As always, if you need further guidance or have any more questions, don't hesitate to get in touch. I'll be back with a more detailed response to your substantive points soon.

Also, other users might have good ideas!!!

**Reply by robintschoetschel** [reply]
Thank you, Rens, for pointing out these resources. The work I'm doing at the moment is for a scoping review, so I'm a bit less concerned about strict reproducibility, and I can err on the side of data storage. In terms of the complications regarding the asynchronous labelling and model training, this setup actually makes things a bit less involved, as the process is more linear. It also saves data, as fewer models are being trained.

**Comment by rohitgarud** [comment]
Hi @robintschoetschel,
I would like to note one thing regarding this workflow. As you are generating a batch of N records at a time and labeling them, you will not be able to take advantage of the ASReview's reranking based on each labeled record in the batch and the classifier will only be trained again after each batch (generally classifier is trained after labeling each record). Depending upon the size N of the batch, this can lead to increased efforts in screening and ASReview will not be utilised to its fullest extent.

**Reply by robintschoetschel** [reply]
Absolutely, it is a less efficient way of doing things. It would be interesting to explore how much of a difference this makes in a simulation study, but that might be a project for a different time.

For me, the biggest advantage of this procedure is that the training of the models is always based on the combined wisdom of the reviewers, and this approach also is a fairly straightforward method for collaboration. I have read some a discussion on collaborating, and to me it seems like this could be an addition to the canon.

**Comment by Rensvandeschoot** [comment]
You might be interested in joining the discussion in this thread: https://github.com/asreview/asreview/issues/282
